tests/getUsersActions.test.js

```diff
@@ -1,74 +1,136 @@
 
-describe("GetUsers module actions", (done) => {
-
-    before(function (done) {
+describe("GetUsers module actions", done => {
+    before(done => {
         const users = [
             {
-                "id": 0,
-                "firstName": "",
-                "lastName": "",
-                "email": "",
-                "phone": ""
+                id: 0,
+                firstName: "",
+                lastName: "",
+                email: "",
+                phone: ""
             }
-        ]
-        
-        writeFile("db/users.json", JSON.stringify(users))
-        .then(() => {
-            done();
-        })
-    })
-
-    it("Should return user with id 0", (done) => {
-        readFile("db/users.json")
-        .then(data => {
-            const users = JSON.parse(data);
-            const userId = 0;
-            const input = findItemById(users, userId);
-            const result = users[0];
-            
-            chai.expect(input).to.deep.equal(result);
+        ];
+
+        writeFile("db/users.json", JSON.stringify(users)).then(() => {
             done();
-        })
+        });
     });
 
-    it("Should return user with id -1", (done) => {
-        readFile("db/users.json")
-        .then(data => {
-            const users = JSON.parse(data);
-            const userId = -1;
-            const input = findItemById(users, userId);
-            const result = undefined;
+    describe("Parse JSON from file function", () => {
+        before(function(done) {
+            writeFile("db/users.json", "[").then(() => {
+                done();
+            });
+        });
 
-            chai.expect(input).to.deep.equal(result);
-            done();
-        })
+        after(function(done) {
+            const users = [
+                {
+                    id: 0,
+                    firstName: "",
+                    lastName: "",
+                    email: "",
+                    phone: ""
+                }
+            ];
+
+            writeFile("db/users.json", JSON.stringify(users)).then(() => {
+                done();
+            });
+        });
+
+        it("Should fail if JSON is invalid", done => {
+            const {
+                parseJSONFromFile
+            } = require("../src/actions/getUsersActions");
+
+            parseJSONFromFile("db/users.json").catch(() => done());
+        });
     });
 
-    it("Should return user with id 'A' ", (done) => {
-        readFile("db/users.json")
-        .then(data => {
-            const users = JSON.parse(data);
-            const userId = 'A';
-            const input = findItemById(users, userId);
-            const result = undefined;
+    describe("Find user by id", () => {
+        it("Should return user with id 0", done => {
+            readFile("db/users.json").then(data => {
+                const users = JSON.parse(data);
+                const userId = 0;
+                const output = findItemById(users, userId);
```

Comment:
**findItemById** e o metoda abstracta care poate fi folosita in mai multe locuri si de aceea numele ei e ok asa
In momentul in care o folosesti, tu stii ce rezultat iti va returna.
Denumeste sugestiv variabila in care salvezi rezultatul.
Ex:
const **foundUser** = findItemById(users, userId);
let **testUser** = users[0];
chai.expect(**foundUser**).to.deep.equal(**testUser**);